### PR TITLE
Feature: Sagemaker compilation jobs

### PR DESCRIPTION
**Pull request by JoshLevyMN**
Adds support for Sagemaker compilation jobs create, describe, list, and stop.

### CODE REVIEW

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @JoshLevyMN, can you resolve the merge conflicts here? The changes look good, so happy to merge when it's ready.

Considering the scope of the changes, the other PR's will need to be rebased as well, but that will probably need to be done one at a time. (I.e., merge this, rebase #7889, merge #7889, rebase next, etc)

**Comment by JoshLevyMN** [comment]
@bblommers this one should be merged now, I'll move on to the next one after this is merged.

**Review by bblommers — APPROVED** [review]
Great, thanks @JoshLevyMN!

**Pull request merged**
